### PR TITLE
fix dhcp::host comment param

### PR DESCRIPTION
**Pull request by jhoblitt**

### CODE REVIEW

File: spec/defines/host_spec.rb [hunk 1]
```diff
@@ -20,7 +20,34 @@
...
+        let :facts do
+          facts
+        end
+
+        let :pre_condition do
+          "class { '::dhcp': interfaces => ['eth0']}"
+        end
+
+        it {
+          verify_concat_fragment_with_comments(catalogue, 'dhcp.hosts+10_myhost.hosts', [
```
Comment (ekohl) [inline spec/defines/host_spec.rb:49]: Can't you just use this?
```ruby
let(:expected_content) do
  <<~CONTENT
...
  CONTENT
end

it {  is_expected.to contain_concat__fragment('dhcp.hosts+10_myhost.hosts').with_content(expected_coontent) }
```

Comment (jhoblitt) [reply spec/defines/host_spec.rb:49]: I don't see why not.  Do you want me to clean it up?

Comment (ehelms) [reply spec/defines/host_spec.rb:49]: Please do, then we can merge and include this in the release being put together for this week.

Comment (jhoblitt) [reply spec/defines/host_spec.rb:49]: Copy. I was only trying to match the style of the existing code.  Will clean it up shortly.

Comment (ekohl) [reply spec/defines/host_spec.rb:49]: I'll be honest and say that a lot is rather old and may have been written with incomplete understanding of rspec or that it simply wasn't that powerful back then.

Comment (jhoblitt) [reply spec/defines/host_spec.rb:49]: I've opened #195 to remove the concat helper methods.  Dealing with the whitespace without add new magic helper methods was miserable.

Comment (jhoblitt) [reply spec/defines/host_spec.rb:49]: I'm planning to rebase and rework this PR if/when #195 is merged.

**Comment by jhoblitt** [comment]
We may want to rebase again after #195 is merged.

File: spec/defines/host_spec.rb [hunk 2]
```diff
@@ -29,6 +29,43 @@
...
+      describe 'comment parameter' do
+        let :params do {
+          :ip      => '10.0.0.100',
+          :mac     => '01:02:03:04:05:06',
+          :comment => 'a useful comment',
+        } end
+
+        let :facts do
+          facts
+        end
```
Comment (ekohl) [inline spec/defines/host_spec.rb:42]: We should pull this to line 6 since it can be shared between both blocks.

**Review by ekohl — COMMENTED** [review]
Feel free to rebase now that it's merged.

File: spec/defines/host_spec.rb (hunk 2 above)
Comment (jhoblitt) [reply spec/defines/host_spec.rb:42]: Great minds...

Comment (jhoblitt) [reply spec/defines/host_spec.rb:50]: Re-pushed.

File: spec/defines/host_spec.rb [hunk 3]
```diff
@@ -19,12 +23,39 @@
...
+
+        let :pre_condition do
+          "class { '::dhcp': interfaces => ['eth0']}"
         end
```
Comment (ekohl) [inline spec/defines/host_spec.rb:54]: Both this `let(:pre_condition)` and `let(:facts)` can be shared by both `describe` blocks by moving them up a level.

Comment (jhoblitt) [reply spec/defines/host_spec.rb]: I've normalized the file.

**Review by ekohl — APPROVED** [review]
Thanks!

**Pull request merged**